cloudpathlib/azure/azblobclient.py

```diff
@@ -155,7 +155,20 @@ def _is_file_or_dir(self, cloud_path: AzureBlobPath) -> Optional[str]:
             return "dir"
 
         try:
-            self._get_metadata(cloud_path)
+            # the result of get_blob_properties is a BlobProperties object (dict mixin)
+            metadata = cast(BlobProperties, self._get_metadata(cloud_path))
+
+            # content_type and content_md5 are never None for files in Azure Blob Storage
+            # if they are None, then the given path is a directory
+            # https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob-properties?tabs=microsoft-entra-id#response-headers
+            is_folder = (
+                metadata.content_settings.content_type is None
+                and metadata.content_settings.content_md5 is None
+            )
```

Comment:
Can you point to a more specific reference? I don't see the link you sent supporting the claim that this is definitive for folders. 

It does seem like `x-ms-resource-type` is useful to answer the question directly, but only for certain configurations so could be included as an optimization. It may be the case that all the scenarios where those vars are none are also ones where `x-ms-resource-type` is available.

Reply:
I agree the reference is sort of vague. I was mainly going of off these two statements:
> Content-Type: The content type that's specified for the blob. If no content type is specified, the default content type is application/octet-stream.

> Content-MD5: If the Content-MD5 header has been set for the blob, this response header is returned so that the client can check for message content integrity.
> In version 2012-02-12 and later, Put Blob sets a block blob’s MD5 value even when the Put Blob request doesn’t include an MD5 header.

Using this, coupled with our observations from real-life usage, we concluded that these parameters are always `None` for folders and never `None` for blobs, even if they are empty.

Reply:
Does your configuration have the `x-ms-resource-type` header for these folder blobs?

Reply:
That header does not seem to be returned within `BlobProperties`, nor can I see it in the in-browser Azure storage explorer.